dlt/destinations/sql_jobs.py

```diff
@@ -660,10 +660,14 @@ def gen_upsert_sql(
                 """)
 
                 # insert records for new elements in the list
-                col_str = ", ".join(["{alias}" + escape_column_id(c) for c in table["columns"]])
+                table_column_names = list(map(escape_column_id, table["columns"]))
+                update_str = ", ".join([c + " = " + "s." + c for c in table_column_names])
+                col_str = ", ".join(["{alias}" + c for c in table_column_names])
                 sql.append(f"""
                     MERGE INTO {table_name} d USING {staging_table_name} s
                     ON d.{unique_column} = s.{unique_column}
+                    WHEN MATCHED
```

Comment:
Not related to `delta` tables.

This fixes a bug in our SQL destination `upsert` implementation. It had slipped trough because we didn't have a test that properly checked record updates. That test is added in this PR.